src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/WPSExecutionManager.java

```diff
@@ -391,6 +391,13 @@ private ExecuteResponseType execute() {
                 SubProgressListener executionListener = new SubProgressListener(listener,
                         inputPercentage, executionPercentage);
                 notifier.checkDismissed();
+                
+                //The system only moves from Queued to Running when the progress notifier fires (!!!).  This fires when inputs are read.
+                //   However, if there are no inputs, it never moves to running state!
+                //Since we are executing, we can move this from queued to running state. 
+                if (status.getPhase() == ProcessState.QUEUED) {
```

Comment:
I have some doubts about this one, the thing is, it reports the service as running even if the thread pools in the process manager cannot actually run it, like, it might stay waiting for an actual start for minutes with the wrong state and progress report at zero... Wondering... could it be better to add a wrapper that forcefully notifies the start of execution before delegating the run method?